.travis.yml

```diff
@@ -38,9 +40,9 @@ addons:
       - qtbase5-dev
       - xvfb
 
-before_install:
+install:
   - if [[ "${TRAVIS_OS_NAME}" = "osx" ]]; then brew update && brew install ninja qt5; fi
 
-script: ./travis.sh 
+script: if [ -z "$COVERALLS_SERVICE_NAME" ]; then ./travis.sh; else ./cover.sh; fi
```

Comment:
Why the separate file? It looks to me like the only necessary difference is that of installing and running lcov followed by submission of the coverage results, that should fit in a single conditional block quite well.